printbuf.c

```diff
@@ -116,20 +151,20 @@ int sprintbuf(struct printbuf *p, const char *msg, ...)
 {
   va_list ap;
   char *t;
-  int size;
-  char buf[128];
+  long int size;
```

Comment:
vsprintf and vasprintf are defined to return int, not long int.  Storing the return value in a long isn't going to be useful.